pkg/broker/broker.go

```diff
@@ -35,46 +36,61 @@ import (
 
 const registrySecretName = "registry-secret"
 
+//go:generate counterfeiter ./ ClusterFactory
+type ClusterFactory interface {
+	DefaultCluster() (k8s.Cluster, error)
+}
+
+//go:generate counterfeiter ./ HelmClientFactory
+type HelmClientFactory interface {
+	HelmClient(cluster k8s.Cluster) my_helm.MyHelmClient
+}
+
+//go:generate counterfeiter ./ ServiceAccountInstallerFactory
+type ServiceAccountInstallerFactory interface {
```

Comment:
Is this the right place for this? I would expect it in the `k8s` package. Also, the generated doubles aren't check in, so ci would fail.

Reply:
I've refactored all the factories into their own files.  Fakes added.  Service creating and deletion works.